utils/frame/try-runtime/cli/src/lib.rs

```diff
@@ -60,31 +56,93 @@ pub struct TryRuntimeCmd {
 		default_value = "Interpreted"
 	)]
 	pub wasm_method: WasmExecutionMethod,
+
+	/// The state to use to run the migration.
+	#[structopt(subcommand)]
+	pub state: State,
 }
 
 /// The state to use for a migration dry-run.
-#[derive(Debug)]
+#[derive(Debug, structopt::StructOpt)]
 pub enum State {
-	/// A snapshot. Inner value is a file path.
-	Snap(String),
+	/// Use a snapshot as state to run the migration.
+	Snap {
+		#[structopt(flatten)]
+		cache_params: CacheParams,
+	},
+
+	/// Use a live chain to run the migration.
+	Live {
+		/// An optional cache file to WRITE to. Not cached if set to `None`.
```

Comment:
@kianenigma I'm not completely sold on the naming here. A cache to me is something that optimize access to some resource and is not necessary for the logic to work; the real resource remains available when the cache is unable to provide the data. This is more of a data file containing state data. 

`state_file`?

Reply:
> A cache to me is something that optimize access to some resource and is not necessary for the logic to work; the real resource remains available when the cache is unable to provide the data. 

The current system also matches your description. In `Live` mode, we always use the remote node, so the resource is there. This is about updating/creating a cache file. 

I am okay with renaming `cache` everywhere to `state_snapshot`.

Reply:
I renamed cache to state_snapshot in remote-externalities. In try-runtime I changed it to snapshot_path because that seems more fitting for the cli. If you like I can change it to state_snapshot as well.